schemas/file.yml

```diff
@@ -25,67 +38,92 @@
   - name: extension
     level: extended
     type: keyword
-    short: File extension.
-    description: >
-      File extension.
-
-      This should allow easy filtering by file extensions.
+    description: File extension.
     example: png
 
   - name: type
     level: extended
     type: keyword
     description: File type (file, dir, or symlink).
+    example: file
 
   - name: device
     level: extended
     type: keyword
     description: Device that is the source of the file.
+    example: sda
+
+  - name: system
+    level: extended
+    type: keyword
+    description: Filesystem used on the volume.
+    example: ntfs, ext4
 
   - name: inode
     level: extended
     type: keyword
     description: Inode representing the file in the filesystem.
+    example: '256383'
 
   - name: uid
     level: extended
     type: keyword
     description: >
       The user ID (UID) or security identifier (SID) of the file owner.
+    example: '1001'
 
   - name: owner
     level: extended
     type: keyword
     description: File owner's username.
+    example: alice
 
   - name: gid
     level: extended
     type: keyword
     description: Primary group ID (GID) of the file.
+    example: '1001'
 
   - name: group
     level: extended
     type: keyword
     description: Primary group name of the file.
+    example: alice
 
   - name: mode
     level: extended
     type: keyword
-    example: 0640
     description: Mode of the file in octal representation.
+    example: '0640'
 
   - name: size
     level: extended
     type: long
-    description: File size in bytes (field is only added when `type` is
-        `file`).
+    short: File size in bytes.
+    description: >
+      File size in bytes.
+
+      Only relevant when `file.type` is "file".
+    example: 16384
+
+  - name: crtime
```

Comment:
This is pretty cryptic for ECS standard I would say? What about `created`?

Reply:
It could also be `creationtime` to keep in line with the `*time` naming schema we have going on.

The `stat` output calls it `Birth` (though always empty on Linux since it does not expose the creation time) - but I don't think that's a good name.

It's maybe unfortunate we already have `ctime` and `mtime` which are UNIX names that Windows does not have. `ctime` could be confused as the creation time. If we ever want to make a change we could go for more descriptive names, e.g. `created|modified|changed|accessed`.

Reply:
Yes, I was specifically pondering this, as I was working on this PR.

The `file` field set didn't get much attention before 1.0.0, unfortunately. But I'm not a fan of the existing field names `ctime` and `mtime`, because they go against one of the principles we're aiming for with ECS. We try to avoid contractions and acronyms, unless the convention is overwhelmingly in favour of them (e.g. `ip`, `pid`).

Seeing that we already had both of these fields, I went with the other 2 standard field names. I went for internal consistency in `file`. But this is getting on my nerves even more, as we discuss this 😂 

As @cwurm points out, `ctime` vs `crtime` are confusing. Any article about Linux filesystem timestamps (debugfs, stat) mention this common confusion.

So I'm very open to revisiting these field names. It's too late for now to change `ctime` and `mtime`, although we could create 4 new fields with less ambiguous names and mark the first two as deprecated for the life of ECS 1.x, and remove them in 2.0.

Proposed names:

- `file.created`: creation time
- `file.accessed`: last access time
- `file.modified` (deprecates `file.mtime`): last modification time for the file content
- `file.metadata_modified` (deprecates `file.ctime`): last modification time for the file metadata.

I'm not a big fan of this last one. Ideas welcome.

For now we can push this naming decision to a bit later, focus on the two new fields, and make a note to deprecate mtime/ctime.